llama_stack/providers/impls/meta_reference/inference/parallel_utils.py

```diff
@@ -23,17 +23,106 @@
     get_model_parallel_src_rank,
 )
 
+from llama_models.llama3.api.datatypes import Message, ToolPromptFormat
+
+from pydantic import BaseModel, Field
+
 from torch.distributed.launcher.api import elastic_launch, LaunchConfig
+from typing_extensions import Annotated
+
+from .generation import TokenResult
+
+
+class InferenceArgs(BaseModel):
+    messages: List[Message]
+    temperature: float
+    top_p: float
+    max_gen_len: int
+    logprobs: bool
```

Comment:
Should this be a bool?

Reply:
Definitely not – it's the pre-existing problem that I raised in chat. I'll cut an issue to track it but will keep it out of scope of this PR so we can land this fix

Reply:
yeah -- this is tracked in https://github.com/meta-llama/llama-stack/issues/214